azb2c-with-parcel.md

```diff
@@ -0,0 +1,106 @@
+```scss
+div {
+  &#api {
+    width: 100px;
+    height: 100px;
+    background-color: #123456;
+  }
+}
+```
+
+Rerunning your developer process should already apply the SCSS style you created.
+Now your set up to design your html/css structure and immediately verify the result in the browser.
+
+### CSS Normalize / Reset
+As still all browsers render unstyled html elements slightly differently, the current web site would result in different appearance on the different browsers, which we have to address.
+We decided to go for [normalize.css](github.com/necolas/normalize.css) as a hard reset felt like an overkill.
+
+### CSS Transpiling
+[Transpiling](https://parceljs.org/languages/css/#transpilation) is done out of the box by parcel.js, so if you have an appropriate configuration in your package.json, no additional steps are required.
+
+## JS
+For our usecase we dont have the need to have any custom JS, plain html & css serves our needs. But parcel.js gives you many options into hand (you could even go for one of the big JS Frameworks and use static site rendering to create your html templates)
+
+## Styling and Testing the login form content
+This far we built a lovely template, but have no idea how the form fields (input for login, pw reset, ...) can be styled. To do that, we need to test our template with the content from the azure b2c flows (and every configuration might have different content).
+For that, access your current available b2c pages and on different screens use the html elements to replace ```<div id="api">...</div>``` to your local en.html. This way you can preview&design those pages before deployment.
+We added a set of those content html elements in separate [*.html files](https://github.com/csalv22/az-b2c-parcel/tree/main/src/api-snippets_de) to quickly retest different scenarios (i.e. styling of error messages).
+
+# Build
+To get a production ready build is fairly simple: Add ```build-b2c: parcel build src/*.html --dist-dir ./public``` to your scripts section in the package.json and test it with ```yarn build-b2c```.
+
+But hold on, if you were to deploy that now and integrate it with azure b2c, this wouldnt work: The azure b2c template mechanism has a restriction, in that it can't handle relative paths from your template like ```<link rel="stylesheet" href="en.6a135913.css"/>``` - it requires absolute paths. So you will need to define the public-url in your build step: ```build-b2c: parcel build src/*.html --dist-dir ./public --public-url 'https://mystorageaccount.blob.core.windows.net/az-b2c'```
+
+The build output can be deployed manually to an [azure blob storage account](https://docs.microsoft.com/en-us/azure/active-directory-b2c/customize-ui-with-html?pivots=b2c-user-flow#2-create-an-azure-blob-storage-account) (or other hosting solution). We deployed it as part of our SPA in an azure static site (behind an app gateway) and integrated it with our CI/CD workflow.
```

Comment:
Would be nice to show it a bit more in detail

Reply:
I agree, but this would be at least one additional blog post and make this one explode in content, to just add "a bit more detail" is difficult.